src/components/AttachmentCarousel/index.js

```diff
@@ -1,386 +1,306 @@
-        });
-        htmlParser.end();
-
-        // Inverting the list for touchscreen devices that can swipe or have an animation when scrolling
-        // promotes the natural feeling of swiping left/right to go to the next/previous image
-        // We don't want to invert the list for desktop/web because this interferes with mouse
-        // wheel or trackpad scrolling (in cases like document preview where you can scroll vertically)
-        if (this.canUseTouchScreen) {
-            attachments.reverse();
-        }
-
-        const page = _.findIndex(attachments, (a) => a.source === this.props.source);
-        if (page === -1) {
-            throw new Error('Attachment not found');
-        }
-
-        // Update the parent modal's state with the source and name from the mapped attachments
-        this.props.onNavigate(attachments[page]);
-
-        return {
-            page,
-            attachments,
-            shouldShowArrow: this.canUseTouchScreen,
-            containerWidth: 0,
-            isZoomed: false,
-            activeSource: null,
-        };
-    }
+            if (newIsZoomed) {
+                toggleArrowsVisibility(false);
+            }
+            setIsZoomed(newIsZoomed);
+        },
+        [isZoomed, toggleArrowsVisibility],
+    );
 
     /**
      * Increments or decrements the index to get another selected item
      * @param {Number} deltaSlide
      */
-    cycleThroughAttachments(deltaSlide) {
-        let delta = deltaSlide;
-        if (this.canUseTouchScreen) {
-            delta = deltaSlide * -1;
-        }
+    const cycleThroughAttachments = useCallback(
+        (deltaSlide) => {
+            let delta = deltaSlide;
+            if (canUseTouchScreen) {
+                delta = deltaSlide * -1;
+            }
 
-        const nextIndex = this.state.page - delta;
-        const nextItem = this.state.attachments[nextIndex];
+            const nextIndex = page - delta;
+            const nextItem = attachments[nextIndex];
 
-        if (!nextItem || !this.scrollRef.current) {
-            return;
-        }
+            if (!nextItem || !scrollRef.current) {
+                return;
+            }
 
-        // The sliding transition is a bit too much on web, because of the wider and bigger images,
-        // so we only enable it for mobile
-        this.scrollRef.current.scrollToIndex({index: nextIndex, animated: this.canUseTouchScreen});
-    }
+            // The sliding transition is a bit too much on web, because of the wider and bigger images,
+            // so we only enable it for mobile
+            scrollRef.current.scrollToIndex({index: nextIndex, animated: canUseTouchScreen});
+        },
+        [attachments, page],
+    );
 
     /**
      * Updates the page state when the user navigates between attachments
      * @param {Array<{item: {source, file}, index: Number}>} viewableItems
      */
-    updatePage({viewableItems}) {
+    const updatePage = useRef(({viewableItems}) => {
```

Comment:
To prevent the `Invariant Violation: Changing onViewableItemsChanged on the fly is not supported` error. We could also go with `useCallback` and include an empty dependency array, so the function never changes, but then we'd have to silence the exhaustive deps rule.